--- cmGlobalXCodeGenerator.cxx.orig	2008-04-17 14:34:37.000000000 +0200
+++ cmGlobalXCodeGenerator.cxx	2008-04-17 14:37:16.000000000 +0200
@@ -1129,9 +1129,10 @@
     }
 
   // Add define flags
-  this->CurrentLocalGenerator->
-    AppendFlags(defFlags,
-                                       this->CurrentMakefile->GetDefineFlags());
+  std::string newDefFlags = "";
+  this->CurrentLocalGenerator->AppendFlags(newDefFlags, this->CurrentMakefile->GetDefineFlags());
+  this->CurrentLocalGenerator->AppendFlags(newDefFlags, defFlags.c_str());
+  defFlags = newDefFlags;
   cmSystemTools::ReplaceString(defFlags, "\"", "\\\"");
   cmSystemTools::ReplaceString(flags, "\"", "\\\"");
   cmSystemTools::ReplaceString(cflags, "\"", "\\\"");


--- cmMakefileTargetGenerator.cxx.orig	2008-04-17 14:54:16.000000000 +0200
+++ cmMakefileTargetGenerator.cxx	2008-04-17 15:03:22.000000000 +0200
@@ -263,19 +263,24 @@
       }
     
     // Add language-specific flags.
+    std::string tmp_flags = "";
     this->LocalGenerator
-      ->AddLanguageFlags(flags, lang,
+      ->AddLanguageFlags(tmp_flags, lang,
                          this->LocalGenerator->ConfigurationName.c_str());
     
     // Add shared-library flags if needed.
-    this->LocalGenerator->AddSharedFlags(flags, lang, shared);
+    this->LocalGenerator->AddSharedFlags(tmp_flags, lang, shared);
     
     // Add include directory flags.
     this->LocalGenerator->
-      AppendFlags(flags, this->LocalGenerator->GetIncludeFlags(lang));
+      AppendFlags(tmp_flags, this->LocalGenerator->GetIncludeFlags(lang));
     // Add include directory flags.
     this->LocalGenerator->
-      AppendFlags(flags,this->GetFrameworkFlags().c_str());
+      AppendFlags(tmp_flags,this->GetFrameworkFlags().c_str());
+
+    this->LocalGenerator->
+      AppendFlags(tmp_flags,flags.c_str());
+    flags = tmp_flags;
 
     *this->FlagFileStream << lang << "_FLAGS = " << flags << "\n\n";
     }
